### DISCUSSION TITLE
Model date format casts no longer being used

### DISCUSSION THREAD

**Discussion by chrisrhymes** [livewire/livewire#6922] [1. Help]
I am updating a project to Livewire 3 and have come across an issue with date formats. I am using legacy_model_binding and have a RegularSchedule model that has opens_at and closes_at with the format `H:i` and valid_from and valid_to that has the date format `Y-m-d` set in the model casts. 

```php
// App\Models\RegularSchedule.php

    protected $casts = [
        'opens_at' => 'date:H:i',
        'closes_at' => 'date:H:i',
        'valid_from' => 'date:Y-m-d',
        'valid_to' => 'date:Y-m-d',
    ];
```

In Livewire 2 it would auto format these into the formats specified in the $casts automatically. In Livewire 3 it now stores them all in the format `Y-m-d H:i:s`. 

Is there a way I can enable the date time format to work as it did previously? Thanks.

**Comment by dragonfly4** [comment]
Just spend a few hours trying to fix the same issue. So I assume it's a bug in v3.0.5

As a workaround I'm setting the format on value level in the Livewire component :unamused:

`$model->valid_from->format('d/m/Y')`
